### DISCUSSION TITLE
Adjusting the third algorithm of `pdsp`

### DISCUSSION THREAD

**Discussion by FanwangM** [theochem/procrustes#189] [Q&A] [answered]
As @banrovegrie mentioned that the third algorithm is not peer-reviewed. Do you think we should adjust a new one? I find this one might be interesting. Thank you. @PaulWAyers 
 [Gillis, N., & Sharma, P. (2018). A semi-analytical approach for the positive semidefinite Procrustes problem. Linear Algebra and its Applications, 540, 112-137.](https://doi.org/10.1016/j.laa.2017.11.023) 

If you want to find the Matlab codes, you will need to navigate to this tab. @banrovegrie 
![Screenshot from 2022-09-02 18-01-14](https://user-images.githubusercontent.com/22950235/188240875-b5d04018-0311-4274-91b3-686da6d129b2.png)

**Comment by banrovegrie** [comment]
Yee, meanwhile I will test the code and confirm the accuracy.

**Comment by PaulWAyers** [comment, marked as answer]
The original "algorithm 3" has been peer reviewed (I think; it is published in a conference proceedings and that is never a sure thing).
     http://www.scielo.org.co/pdf/rcm/v55n1/0034-7426-rcm-55-01-109.pdf

However, I really like Fanwang's suggestion. The Oviedo paper (algorithm 3) performs similarly to what they call `PARATAN` and what the above reference calls `PARTAN`. The above reference seems to have good performance, and has four closely related algorithms which could be implemented simulataneously, switching between the algorithms as specified by the user by a keyword. This is explained at the top of Section 5.

Specifically, the 
- projected gradient method
- fast gradient method
- parallel tangents method
- semi-analytic fast gradient method

This gives a nice alternative to the first two algorithms, which focus on a more analytic approach (more like the other methods we have considered). This is a numerical approach, and perhaps might be faster (hard to know until we test).
